sound/soc/codecs/tfa989x.c

```
@@ -239,10 +244,28 @@ static int tfa989x_i2c_probe(struct i2c_client *i2c)
 {
 	struct device *dev = &i2c->dev;
 	const struct tfa989x_rev *rev;
+	struct tfa989x_priv *tfa989x;
 	struct regmap *regmap;
 	unsigned int val;
 	int ret;
 
+	tfa989x = devm_kzalloc(dev, sizeof(*tfa989x), GFP_KERNEL);
+	if (!tfa989x)
+		return -ENOMEM;
+
+	tfa989x->vddd = devm_regulator_get(dev, "vddd");
+	if (IS_ERR(tfa989x->vddd)) {
+		if (PTR_ERR(tfa989x->vddd) == -EPROBE_DEFER)
+			return -EPROBE_DEFER;
+		tfa989x->vddd = NULL;
```

I don't think it's necessary to ignore errors here, can use:

```c
if (IS_ERR(...))
	return dev_err_probe(...);
```